### PR TITLE
Prevent warning on wc_get_product_variation_attributes

### PR DESCRIPTION
**Pull request by Rahmon**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Since [get_post_meta](https://github.com/woocommerce/woocommerce/blob/14a8b328a975a22b7d92602f471d27b06ade173d/plugins/woocommerce/includes/wc-product-functions.php#L689) can return `false` when the input is a boolean, zero or string, this PR handles this case to ensure that the return of get_post_meta is an array. This is important because the `$all_meta` is used in a [foreach](https://github.com/woocommerce/woocommerce/blob/14a8b328a975a22b7d92602f471d27b06ade173d/plugins/woocommerce/includes/wc-product-functions.php#L707) and prevents the warning `Invalid argument supplied for foreach()`.

### How to test the changes in this Pull Request:

1. Enable `debug.log` by adding the following to `wp-config.php`
```
define( 'WP_DEBUG', true );
define( 'WP_DEBUG_LOG', true );
define( 'WP_DEBUG_DISPLAY', false );
```
2. Run 
```
wp shell
wp> wc_get_product_variation_attributes( 0 )
```
3. You should not see a warning either in wp shell or in `debug.log`

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Dev - Prevent warning on wc_get_product_variation_attributes

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/includes/wc-product-functions.php [hunk 1]
```diff
@@ -686,7 +686,7 @@ function wc_get_product_id_by_sku( $sku ) {
  */
 function wc_get_product_variation_attributes( $variation_id ) {
 	// Build variation data from meta.
-	$all_meta                = get_post_meta( $variation_id );
+	$all_meta                = array_filter( (array) get_post_meta( $variation_id ) );
```
Comment (vedanshujain) [inline plugins/woocommerce/includes/wc-product-functions.php]: we should change this to use the `is_array` check, there may be cases where an empty value for a meta is meaningful information

**Review by vedanshujain — DISMISSED** [review]
hey @Rahmon , thanks for the PR. requesting a minor change before we merge

File: plugins/woocommerce/includes/wc-product-functions.php (hunk 1 above)
Comment (Rahmon) [reply plugins/woocommerce/includes/wc-product-functions.php]: @vedanshujain thanks for reviewing my PR. :pray: 

It makes sense to me and I've updated the code.

**Comment by rrennick** [comment]
@Rahmon I apologize for the long delay in responding to this PR. Could you merge in the latest trunk?

**Comment by Rahmon** [comment]
Hello @rrennick,

I've updated the PR. Thanks.

**Comment by rrennick** [comment]
@Rahmon Thanks for the quick follow up. This looks great. I wasn't able to add a changelog file to the branch. Could you add the following to `plugins/woocommerce/changelog/pr-32288`:

```
Significance: patch
Type: fix

Fixed warning on wc_get_product_variation_attributes when product does not exist
```

**Comment by Rahmon** [comment]
> @Rahmon Thanks for the quick follow up. This looks great. I wasn't able to add a changelog file to the branch. Could you add the following to `plugins/woocommerce/changelog/pr-32288`:
> 
> ```
> Significance: patch
> Type: fix
> 
> Fixed warning on wc_get_product_variation_attributes when product does not exist
> ```

@rrennick 

Added :heavy_check_mark:

**Review by rrennick — APPROVED** [review]
@Rahmon Thanks for the help.

**Pull request merged**